### PR TITLE
Remove Object from VoxelComplex

### LINKED ISSUE #1347: testVoxelComplex too long
The `testVoxelComplex` unit test is too long (40sec on my machine).
We should check if all tests are really necessary..

### PR DESCRIPTION
**Pull request by phcerdan**
This drops the runtime by ~4x (orientative)
And the memory usage (not quantified).

Also reduce time of testing. Fix #1347

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [NA] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

**Comment by phcerdan** [comment]
I fixed CI errors. Travis-ci is red because compiling timeout, CMake failing to configure with doxygen, and the DGtal Tools is failing, but PR opened at: https://github.com/DGtal-team/DGtalTools/pull/345.

**Comment by dcoeurjo** [comment]
PR pending for doxygen (#1372). Thx

**Comment by phcerdan** [comment]
How can I help to move this forward? I have a few scripts pending on this update.
Thanks!

**Comment by dcoeurjo** [comment]
Hi @phcerdan can you please merge the master to this branch ? All travis issues should disappear.

**Comment by phcerdan** [comment]
Will do after new year, thanks!

**Comment by phcerdan** [comment]
done, rebased on top of master.

File: src/DGtal/topology/VoxelComplex.h [hunk 1]
```diff
@@ -240,7 +242,7 @@ class VoxelComplex : public CubicalComplex<TKSpace, TCellContainer> {
      * Insert cell (voxel) in the khalimsky space AND in the object set.
      *
      * @param kcell input voxel
-     * @param close_it if true, apply @ref voxelClose.
+     * @param close_it if true, apply @voxelClose.
```
Comment (dcoeurjo) [inline src/DGtal/topology/VoxelComplex.h]: why did you remove the @ref ? Looks like it would produce Doxygen errors..

File: src/DGtal/topology/VoxelComplex.h [hunk 2]
```diff
@@ -250,39 +252,40 @@ class VoxelComplex : public CubicalComplex<TKSpace, TCellContainer> {
      * Insert cell(voxel) in K-space and in the object set.
      *
      * @param data_pair pair<Cell, Data>
-     * @param close_it if true, apply @ref voxelClose
+     * @param close_it if true, apply @voxelClose
```
Comment (dcoeurjo) [inline src/DGtal/topology/VoxelComplex.h]: Same here

File: src/DGtal/topology/VoxelComplex.h [hunk 3]
```diff
@@ -250,39 +252,40 @@ class VoxelComplex : public CubicalComplex<TKSpace, TCellContainer> {
...
     }
 
     /**
-     * Create a uSpel from the input Point and insert it using insertVoxelCell.
+     * Create a @uSpel from the input Point and insert it using insertVoxelCell.
      * @see insertVoxelCell
      *
      * @param dig_point input point of the KSpace
-     * @param close_it flag to apply @ref voxelClose
+     * @param close_it flag to apply @voxelClose
```
Comment (dcoeurjo) [inline src/DGtal/topology/VoxelComplex.h]: Again

File: src/DGtal/topology/VoxelComplex.h [hunk 4]
```diff
@@ -390,7 +391,7 @@ class VoxelComplex : public CubicalComplex<TKSpace, TCellContainer> {
 
     /**
      * Return all critical cliques for \b cubical.
-     * It calls criticalCliquesForD()
+     * It calls @criticalCliquePairForD
```
Comment (dcoeurjo) [inline src/DGtal/topology/VoxelComplex.h]: Same here... Maybe that's a Doxygen feature ;).. I need to check..

File: src/DGtal/topology/VoxelComplex.ih [hunk 5]
```diff
@@ -227,38 +182,47 @@ inline void DGtal::VoxelComplex<TKSpace, TObject, TCellContainer>::cellsClose(
...
+
 //---------------------------------------------------------------------------
-template <typename TKSpace, typename TObject, typename TCellContainer>
-inline void DGtal::VoxelComplex<TKSpace, TObject, TCellContainer>::clear() {
+template <typename TKSpace, typename TCellContainer>
+template <typename TDigitalSet>
+inline void
+DGtal::VoxelComplex<TKSpace, TCellContainer>::dumpVoxels(TDigitalSet & in_out_set) {
+    auto &ks = this->space();
+    for (auto it = this->begin(3), itE = this->end(3) ; it != itE ; ++it ){
```
Comment (dcoeurjo) [inline src/DGtal/topology/VoxelComplex.ih]: what do the `begin(3)` and `end(3)` stand for?

File: src/DGtal/topology/VoxelComplex.h (hunk 1 above)
Comment (dcoeurjo) [reply src/DGtal/topology/VoxelComplex.h]: Not working apparently. You should keep the @ref.

capture d ecran 2019-01-05 a 17 33 22

Comment (phcerdan) [reply src/DGtal/topology/VoxelComplex.h]: Thanks, I will fix it.

Comment (phcerdan) [reply src/DGtal/topology/VoxelComplex.h]: @ref's are back, thanks

Comment (dcoeurjo) [reply src/DGtal/topology/VoxelComplex.h]: oops my bad.. You forced-pushed :)
Many thanks for this great PR !
merging

**Comment by phcerdan** [comment]
Thanks @dcoeurjo! :tada:
